### PR TITLE
feat: use mydata-e2e-simulator instead of jest.mock

### PR DESCRIPTION
**Pull request by Graborg**
**Ticket:** [#22](https://trello.com/c/QiBf91a6/22-remove-mocking-of-mydata-in-integration-tests)

**Intent:**
Stop mocking myDate in integration tests.
Now uses myData's route for requesting and giving consent, without the involvement of any app.
It uses 2 images for the setup. One operator and one e2e-simulator For now, you will have to build them yourself from the myData repo.
https://github.com/JobtechSwe/mydata/blob/master/app/Dockerfile
tagged as e2e-simulator, **Important First bump the node version to 10.5. Since the flow won't work without it.**
https://github.com/JobtechSwe/mydata/blob/master/operator/Dockerfile
tagged as operator
**How to test :**

run `docker-compose up`

- [x] I've written tests for the code. 

### All of the following commands should pass.

- [x] `npm test`
- [x] `npm run integration`

### CODE REVIEW

File: Dockerfile.test [hunk 1]
```diff
@@ -0,0 +1,18 @@
+FROM node:11.7-alpine
```
[the remark this answers is not in the document]
Comment (Graborg) [reply Dockerfile.test]: Although, our Dockerfile says 11.7. I think it makes sense to keep the versions in the Dockerfiles the same. If we want to upgrade I think that should be done in a separate PR.

File: docker-compose.yml [hunk 2]
```diff
@@ -21,19 +36,33 @@ services:
...
+      - NODE_ENV=development
       - PGHOST=postgres
       - PGUSER=postgresuser
       - PGPASSWORD=postgrespassword
       - PGDATABASE=mydata
       - REDIS=redis://redis-api:6379/
+    command: bash -c "npm run migrate up && npm run start"
+
+  e2e-simulator: #This replaces the phone app in the normal flow.
+    image: e2e-simulator
```
[the remark this answers is not in the document]
Comment (Graborg) [reply docker-compose.yml]: Yeah, I agree a better name is needed. But in the end that should probably be decided by MyData team. And hopefully we can get their image directly.  mydata-app-simulator to me sounds very much alike the simulator that we actually build for IOS/Android 😄 .
How about; mydata-e2e-app-simulator?

File: docker-compose.yml [hunk 3]
```diff
@@ -8,6 +22,7 @@ services:
           - redis-api
     ports:
       - 6380:6379
+
   postgres:
     image: postgres:9.6
```
[the remark this answers is not in the document]
Comment (Graborg) [reply docker-compose.yml]: 😄 Sure. Lets do it.

File: docker-compose.yml [hunk 4]
```diff
@@ -21,19 +36,33 @@ services:
...
+  e2e-simulator: #This replaces the phone app in the normal flow.
+    image: e2e-simulator
+    environment:
+      - OPERATOR_URL=http://operator:4000/api
+    networks:
+      mydata:
+        aliases:
+          - e2e-simulator
 networks:
   mydata:
```
[the remark this answers is not in the document]
Comment (Graborg) [reply docker-compose.yml]: 👍

File: docker-compose.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (lundeeell) [reply docker-compose.yml]: well, then I think mydata-operator sounds better, as it is exactly that, and doesn't really have anything to do with just e2e :)

[the remark this answers is not in the document]
Comment (Graborg) [reply docker-compose.yml]: No, this is specifically designed for e2e.

File: docker-compose.yml [hunk 5]
```diff
@@ -1,5 +1,19 @@
 version: '3'
 services:
+  myskills:
+    build:
+      context: .
+      dockerfile: Dockerfile.test
```
[the remark this answers is not in the document]
Comment (Graborg) [reply docker-compose.yml]: ✅

File: Dockerfile.test (hunk 1 above)
[the remark this answers is not in the document]
Comment (Graborg) [reply Dockerfile.test]: ✅

**Comment by Graborg** [comment]
...To support startup order (wait for postgres) I de-bumped docker-compose version in docker compose test to '2.4'

**Pull request merged**